### PR TITLE
Ianhelle/mp dependencies 2020 12 07

### PR DESCRIPTION
**Pull request by ianhelle**
Refactored dependencies into extras.
Note this is based on the Pivot branch so has all of the commits included - these will disappear when that branch is merged

### CODE REVIEW

File: docs/source/getting_started/Installing.rst [hunk 1]
```diff
@@ -17,25 +17,157 @@ with many of packages required by *msticpy* pre-installed.
...
+Extras in msticpy
+~~~~~~~~~~~~~~~~~
+
+The extras available in *msticpy* are described in the following table:
+
++------------------+------------------------------------+--------------+--------------+
+| extra            | Functionality                      | Install time | Install time |
+|                  |                                    | (increment)  | (full)       |
++==================+====================================+==============+==============+
+| default install  | - Most functionality (approx 75%)  |       --     |   1m:13s     |
```
Comment (petebryan) [inline docs/source/getting_started/Installing.rst:90]: I think we should explicitly call out that this won't install KQL stuff. Given KQL is the core of the notebooks we have produced so far I think its worth highlighting this.

Comment (petebryan) [reply docs/source/getting_started/Installing.rst:90]: I know you say this below but if other people are like me they will just jump straight to the table.

File: docs/source/getting_started/Installing.rst [hunk 2]
```diff
@@ -17,25 +17,157 @@ with many of packages required by *msticpy* pre-installed.
...
+   :height: 3in
+
+To fix this simply run pip install with the "extra" option shown in the
+exception message:
+
+.. code:: bash
+
+    pip install msticpy[ml]
+
+.. note:: In some cases you many not get an informative error. (We've
```
Comment (petebryan) [inline docs/source/getting_started/Installing.rst:168]: Errant (

File: docs/source/getting_started/Installing.rst [hunk 3]
```diff
@@ -17,25 +17,157 @@ with many of packages required by *msticpy* pre-installed.
...
+   tried to trap all of the cases but if
+   experience a problem with some *msticpy* functionality (especially
```
Comment (petebryan) [inline docs/source/getting_started/Installing.rst:170]: Open ended (

**Review by petebryan — APPROVED** [review]

File: docs/source/getting_started/Installing.rst (hunk 1 above)
Comment (ianhelle) [reply docs/source/getting_started/Installing.rst]: I've put a big warning before this table

File: docs/source/getting_started/Installing.rst (hunk 3 above)
Comment (ianhelle) [reply docs/source/getting_started/Installing.rst]: Fixed

**Review by petebryan — APPROVED** [review]

**Pull request merged**